src/AppInstallerRepositoryCore/Rest/Schema/1_0/Interface.cpp

```diff
@@ -0,0 +1,187 @@
+
+    // General API response constants
+    constexpr std::string_view Data = "data"sv;
+
+    // Search body constants
+    constexpr std::string_view FetchAllManifests = "fetchAllManifests"sv;
+
+    // Search response constants
+    constexpr std::string_view PackageIdentifier = "PackageIdentifier"sv;
+    constexpr std::string_view PackageName = "PackageName"sv;
+    constexpr std::string_view Publisher = "Publisher"sv;
+    constexpr std::string_view PackageFamilyName = "PackageFamilyName"sv;
+    constexpr std::string_view ProductCode = "ProductCode"sv;
+    constexpr std::string_view Versions = "Versions"sv;
+    constexpr std::string_view Version = "version"sv;
+    constexpr std::string_view Channel = "Channel"sv;
+
+    namespace
+    {
+        utility::string_t GetJsonKeyNameString(std::string_view nodeName)
+        {
+            return utility::conversions::to_string_t(nodeName.data());
+        }
+
+        web::json::value GetSearchBody(const SearchRequest& searchRequest)
+        {
+            // TODO: Use search request to construct search body.
+            UNREFERENCED_PARAMETER(searchRequest);
+
+            web::json::value json_body;
+            json_body[GetJsonKeyNameString(FetchAllManifests)] = web::json::value::string(L"true");
+
+            return json_body;
+        }
+
+        std::string GetRestAPIBaseUri(std::string restApiUri)
+        {
+            if (restApiUri.back() == '/')
+            {
+                restApiUri.pop_back();
+            }
+
+            return restApiUri;
+        }
+
+        utility::string_t GetSearchEndpoint(const std::string& restApiUri)
+        {
+            std::string fullSearchAPI = restApiUri;
+            return utility::conversions::to_string_t(fullSearchAPI.append(ManifestSearchPostEndpoint));
+        }
+
+        utility::string_t GetManifestByVersionEndpoint(
+            const std::string& restApiUri, const std::string& packageId, const std::string& version, const std::string& channel)
+        {
+            std::string versionEndpoint = restApiUri;
+            versionEndpoint.append(ManifestByVersionAndChannelGetEndpoint).append(packageId);
+
+            // Add Version Query param
+            if (!version.empty())
+            {
+                versionEndpoint.append("?version=").append(version);
+            }
+
+            // Add Channel Query param
+            if (!version.empty())
+            {
+                versionEndpoint.append("&channel=").append(channel);
+            }
+
+            return utility::conversions::to_string_t(versionEndpoint);
+        }
+
+        std::optional<std::string> GetStringFromJsonStringValue(const web::json::value& value)
+        {
+            if (value.is_null() || !value.is_string())
+            {
+                return {};
+            }
+
+            std::string result = Utility::Trim(utility::conversions::to_utf8string(value.as_string()));

```

Comment:
>Trim [](start = 42, length = 4)

Why are you trimming the value?  Shouldn't it just return whatever is defined in the JSON?  Also, why are you treating and empty value as not present.  That doesn't allow for intentionally empty values.

I would suggest a method that does the basic json::value to string converion, then another one that applies these additional semantics with a name that implies it is doing so.